// @liripeng/vue-audio-player/lib/vue-audio-player.es.js
!function(e2) {
  var t, c2, l2, o2, n2, i2, a2 = '<svg><symbol id="icon-play-volume-no" viewBox="0 0 1024 1024"><path d="M529.1008 901.60128c-2.69824 0-12.09856-4.10112-16.9984-9.00096l-0.30208-0.30208-230.4-190.10048H96.1024c-8.89856 0-18.29888-4.59776-22.70208-9.00096s-9.00096-13.7984-9.00096-22.70208V351.49824c0-6.99904 2.69824-12.9024 4.39808-14.60224h2.2016l2.40128-2.40128c4.39808-4.39808 13.7984-9.00096 22.70208-9.00096h185.29792l230.4-195.79904 0.19968-0.19456c5.80096-5.80096 12.9024-6.99904 17.80224-6.99904 4.39808 0 8.89856 0.9984 12.60032 2.89792l0.50176 0.30208 0.50176 0.19968c12.49792 4.1984 17.19808 11.29984 17.19808 26.39872v717.7984c0 15.09888-4.70016 22.2976-17.19808 26.39872l-1.80224 0.59904-1.39776 1.39776c-3.00544 3.00544-5.6064 3.10784-11.10528 3.10784z"  ></path></symbol><symbol id="icon-play-volume" viewBox="0 0 1024 1024"><path d="M529.1 901.6c-2.7 0-12.1-4.1-17-9l-0.3-0.3-230.4-190.1H96.1c-8.9 0-18.3-4.6-22.7-9-4.4-4.4-9-13.8-9-22.7v-319c0-7 2.7-12.9 4.4-14.6H71l2.4-2.4c4.4-4.4 13.8-9 22.7-9h185.3l230.4-195.8 0.2-0.2c5.8-5.8 12.9-7 17.8-7 4.4 0 8.9 1 12.6 2.9l0.5 0.3 0.5 0.2c12.5 4.2 17.2 11.3 17.2 26.4v717.8c0 15.1-4.7 22.3-17.2 26.4l-1.8 0.6-1.4 1.4c-3 3-5.6 3.1-11.1 3.1z m279.1-79.7c-9.9 0-23.1-5.1-26.9-12.6-9-18-5.8-36.4 7.9-46.1 4.8-2 11.6-7.2 20.3-15.6 8.9-8.7 22.4-23.7 36.2-45.7 23-36.8 50.4-99.7 50.4-190.9s-29-154.2-53.3-191.1c-14.5-22-28.8-37-38.2-45.6-6.6-6.1-14.9-13-21-15.5-12.6-9.1-17-31.4-8.9-44.5 9.4-9.1 20.6-14.3 30.8-14.3 5.4 0 10.4 1.4 14.8 4.2 0.7 0.6 1.6 1.3 3 2.4 29.4 23.1 54.9 51.4 75.8 84.1 40.1 62.9 60.5 137 60.5 220.3 0 83.7-19.7 158.1-58.4 221.1-20.1 32.7-44.5 60.9-72.6 83.6-1.7 1.4-2.6 2.1-3.3 2.8-3.4 3.4-13.3 3.4-17.1 3.4zM688.6 696.6c-8.3 0-22.6-9.7-26.9-18.3l-0.2-0.5-0.3-0.4c-8.2-12.4 0.8-30.4 14.5-39.7 6.4-3.4 60.9-35.5 60.9-132.3 0-46.5-18-78.4-33.2-97-16.5-20.2-33.1-29.4-33.7-29.8l-0.6-0.3-0.7-0.2c-5.8-1.9-11.4-8.5-14.3-16.8-2.9-8.3-2.3-16.6 1.4-22.2l0.6-0.9 0.3-1c2.9-8.6 15.6-16.1 27.3-16.1 4.5 0 8.6 1.1 11.7 3.2l2.1 1.4h1.5c4.5 1.7 29.1 14 53.5 41.9 21.7 24.9 47.6 68.1 47.6 132.2 0 72.9-24.5 120.2-45 147.1-22.6 29.5-45.6 42.2-50.4 44.1h-2.4l-2.4 2.4c-3.2 3-5.7 3.2-11.3 3.2z"  ></path></symbol><symbol id="icon-play-pause" viewBox="0 0 1024 1024"><path d="M960 512c0-247.36-200.576-448-448-448C264.64 64 64 264.64 64 512c0 247.424 200.64 448 448 448C759.424 960 960 759.424 960 512zM128 511.936c0-212.032 171.904-384 384-384s384 171.968 384 384-171.904 384-384 384S128 723.968 128 511.936zM406.784 678.208c-22.656 0.064-41.088-19.072-41.088-42.56L365.696 383.04c0-23.68 18.432-42.688 41.152-42.688C429.568 340.224 448 359.488 448 382.976l0 252.544C448 659.136 429.568 678.272 406.784 678.208zM617.088 678.208C594.368 678.272 576 659.136 576 635.648L576 383.04c0-23.68 18.432-42.688 41.152-42.688 22.656-0.128 41.152 19.136 41.152 42.624l0 252.544C658.24 659.136 639.808 678.272 617.088 678.208z"  ></path></symbol><symbol id="icon-play" viewBox="0 0 1024 1024"><path d="M511.2269 64.4608c-246.1655 0-446.4425 200.278-446.4425 446.4425 0 246.1655 200.277 446.4435 446.4425 446.4435s446.4425-200.278 446.4425-446.4435C957.6694 264.7388 757.3914 64.4608 511.2269 64.4608zM511.2269 901.5409c-215.4015 0-390.6376-175.2361-390.6376-390.6376 0-215.4015 175.2361-390.6365 390.6376-390.6365 215.4015 0 390.6365 175.2351 390.6365 390.6365C901.8634 726.3048 726.6273 901.5409 511.2269 901.5409zM705.3455 459.649l-238.807-129.622c-11.3347-6.186-23.5151-9.4556-35.2328-9.4556-34.5508 0-59.5927 26.5134-59.5927 63.0272v254.61248c0 36.5117 25.0419 62.9975 59.5927 63.0262 11.7166 0 23.8971-3.2696 35.1519-9.3993l240.087-129.7326c21.718-11.8262 34.1985-30.2735 34.1985-50.5743C740.7432 490.8216 727.852 471.9114 705.3455 459.649zM679.9227 513.0568 439.8356 642.7884c-4.1974 2.3183-7.168 2.6429-8.5299 2.6429-1.5247 0-3.7878 0-3.7878-7.2212V383.59859199999994c0-7.2212 2.263-7.2212 3.7878-7.2212 1.3629 0 4.3325 0.3543 8.5555 2.6429l238.8357 129.6507c1.7172 0.9533 3.0515 1.8524 4.0878 2.6429C682.0219 511.8587 681.0665 512.4577 679.9227 513.0568z"  ></path></symbol><symbol id="icon-play-prev" viewBox="0 0 1024 1024"><path d="M128 102.4V928c0 17.673 14.327 32 32 32 17.673 0 32-14.327 32-32V102.4c0-17.673-14.327-32-32-32-17.673 0-32 14.327-32 32z m646.028 41.395l-404.352 331.09c-27.383 22.42-27.435 58.886 0 81.35l404.352 331.09c27.382 22.42 49.676 12.003 49.676-23.366V167.16c0-35.646-22.24-45.83-49.676-23.365z m-40.546-49.518c69.12-56.597 154.222-16.897 154.222 72.883v696.8c0 89.481-85.14 129.447-154.222 72.883L329.13 605.753c-58.7-48.064-58.662-132.353 0-180.386l404.352-331.09z" fill="" ></path></symbol><symbol id="icon-play-next" viewBox="0 0 1024 1024"><path d="M889.6 921.6V96c0-17.673-14.327-32-32-32-17.673 0-32 14.327-32 32v825.6c0 17.673 14.327 32 32 32 17.673 0 32-14.327 32-32z m-646.028-41.395l404.352-331.09c27.383-22.42 27.435-58.886 0-81.35l-404.352-331.09c-27.382-22.42-49.676-12.003-49.676 23.366V856.84c0 35.646 22.24 45.83 49.676 23.365z m40.546 49.518c-69.12 56.597-154.222 16.897-154.222-72.883v-696.8c0-89.481 85.14-129.447 154.222-72.883l404.352 331.09c58.7 48.064 58.662 132.353 0 180.386l-404.352 331.09z" fill="" ></path></symbol></svg>', d2 = (d2 = document.getElementsByTagName("script"))[d2.length - 1].getAttribute("data-injectcss");
  if (d2 && !e2.__iconfont__svg__cssinject__) {
    e2.__iconfont__svg__cssinject__ = true;
    try {
      document.write("<style>.svgfont {display: inline-block;width: 1em;height: 1em;fill: currentColor;vertical-align: -0.1em;font-size:16px;}</style>");
    } catch (e3) {
      console && console.log(e3);
    }
  }
  function s2() {
    n2 || (n2 = true, l2());
  }
  t = function() {
    var e3, t2, c3, l3;
    (l3 = document.createElement("div")).innerHTML = a2, a2 = null, (c3 = l3.getElementsByTagName("svg")[0]) && (c3.setAttribute("aria-hidden", "true"), c3.style.position = "absolute", c3.style.width = 0, c3.style.height = 0, c3.style.overflow = "hidden", e3 = c3, (t2 = document.body).firstChild ? (l3 = e3, (c3 = t2.firstChild).parentNode.insertBefore(l3, c3)) : t2.appendChild(e3));
  }, document.addEventListener ? ~["complete", "loaded", "interactive"].indexOf(document.readyState) ? setTimeout(t, 0) : (c2 = function() {
    document.removeEventListener("DOMContentLoaded", c2, false), t();
  }, document.addEventListener("DOMContentLoaded", c2, false)) : document.attachEvent && (l2 = t, o2 = e2.document, n2 = false, (i2 = function() {
    try {
      o2.documentElement.doScroll("left");
    } catch (e3) {
      return void setTimeout(i2, 50);
    }
    s2();
  })(), o2.onreadystatechange = function() {
    o2.readyState == "complete" && (o2.onreadystatechange = null, s2());
  });
}(window);
/*! *****************************************************************************
Copyright (c) Microsoft Corporation.

Permission to use, copy, modify, and/or distribute this software for any
purpose with or without fee is hereby granted.

THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
PERFORMANCE OF THIS SOFTWARE.
***************************************************************************** */
var extendStatics = function(d2, b2) {
  extendStatics = Object.setPrototypeOf || { __proto__: [] } instanceof Array && function(d3, b3) {
    d3.__proto__ = b3;
  } || function(d3, b3) {
    for (var p in b3)
      if (Object.prototype.hasOwnProperty.call(b3, p))
        d3[p] = b3[p];
  };
  return extendStatics(d2, b2);
};
function __extends(d2, b2) {
  if (typeof b2 !== "function" && b2 !== null)
    throw new TypeError("Class extends value " + String(b2) + " is not a constructor or null");
  extendStatics(d2, b2);
  function __() {
    this.constructor = d2;
  }
  d2.prototype = b2 === null ? Object.create(b2) : (__.prototype = b2.prototype, new __());
}
var __assign = function() {
  __assign = Object.assign || function __assign2(t) {
    for (var s2, i2 = 1, n2 = arguments.length; i2 < n2; i2++) {
      s2 = arguments[i2];
      for (var p in s2)
        if (Object.prototype.hasOwnProperty.call(s2, p))
          t[p] = s2[p];
    }
    return t;
  };
  return __assign.apply(this, arguments);
};
function __rest(s2, e2) {
  var t = {};
  for (var p in s2)
    if (Object.prototype.hasOwnProperty.call(s2, p) && e2.indexOf(p) < 0)
      t[p] = s2[p];
  if (s2 != null && typeof Object.getOwnPropertySymbols === "function")
    for (var i2 = 0, p = Object.getOwnPropertySymbols(s2); i2 < p.length; i2++) {
      if (e2.indexOf(p[i2]) < 0 && Object.prototype.propertyIsEnumerable.call(s2, p[i2]))
        t[p[i2]] = s2[p[i2]];
    }
  return t;
}
function __values(o2) {
  var s2 = typeof Symbol === "function" && Symbol.iterator, m2 = s2 && o2[s2], i2 = 0;
  if (m2)
    return m2.call(o2);
  if (o2 && typeof o2.length === "number")
    return {
      next: function() {
        if (o2 && i2 >= o2.length)
          o2 = void 0;
        return { value: o2 && o2[i2++], done: !o2 };
      }
    };
  throw new TypeError(s2 ? "Object is not iterable." : "Symbol.iterator is not defined.");
}
var e$1 = function() {
  function e2() {
    this.__map = {};
  }
  return e2.prototype.beforeEach = function(t) {
    this.__interceptor = t;
  }, e2.prototype.on = function(r2, e3) {
    var i2, o2, n2 = Array.isArray(r2) ? r2 : [r2];
    try {
      for (var a2 = __values(n2), _ = a2.next(); !_.done; _ = a2.next()) {
        var p = _.value;
        this.__map[p] = this.__map[p] || [];
        var f2 = this.__map[p];
        f2 && f2.push(e3);
      }
    } catch (t) {
      i2 = { error: t };
    } finally {
      try {
        _ && !_.done && (o2 = a2.return) && o2.call(a2);
      } finally {
        if (i2)
          throw i2.error;
      }
    }
    return this;
  }, e2.prototype.emit = function(t, r2, e3) {
    var i2 = this;
    this.__interceptor !== void 0 ? this.__interceptor(t, function() {
      i2.__emit(t, r2), e3 && e3();
    }) : (this.__emit(t, r2), e3 && e3());
  }, e2.prototype.__emit = function(e3, i2) {
    var o2, n2, a2 = this.__map[e3];
    if (Array.isArray(a2) && (a2 == null ? void 0 : a2.length))
      try {
        for (var _ = __values(a2), p = _.next(); !p.done; p = _.next()) {
          (0, p.value)(i2);
        }
      } catch (t) {
        o2 = { error: t };
      } finally {
        try {
          p && !p.done && (n2 = _.return) && n2.call(_);
        } finally {
          if (o2)
            throw o2.error;
        }
      }
    this.event = __assign(__assign({}, i2), { __type: e3 });
  }, e2.prototype.off = function(t, r2) {
    var e3 = this.__map[t];
    if (e3 !== void 0)
      if (r2 === void 0)
        delete this.__map[t];
      else {
        var i2 = e3.findIndex(function(t2) {
          return t2 === r2;
        });
        e3.splice(i2, 1);
      }
  }, e2.prototype.destroy = function() {
    this.__map = {};
  }, e2;
}();
var e = Object.prototype.toString;
function n(t) {
  return e.call(t) === "[object Function]";
}
var o, c = "clientX", u = "clientY", r = 16, a$1 = "start", i$1 = "move", d$2 = "cancel", l = "end", m$1 = "computed", s = "left", v$1 = "right", f$1 = "up", h$1 = "down", g = "touchstart", j = "touchmove", w$1 = "touchend", x$1 = "touchcancel", y = "mouseup", F = "mousemove", M = "mousedown", N = ((o = {})[4] = "start", o[5] = "move", o[1] = "end", o[3] = "cancel", o);
function O(t) {
  return N[t];
}
function S(t, e2, n2) {
  var o2, c2, u2, r2, a2, i2, d2, l2 = { 1: (o2 = {}, o2[0] = (c2 = {}, c2.move = 4, c2), o2[4] = (u2 = {}, u2.move = 5, u2.end = 1, u2.cancel = 3, u2), o2[5] = (r2 = {}, r2.move = 5, r2.end = 1, r2.cancel = 3, r2), o2), 0: (a2 = {}, a2[4] = (i2 = {}, i2.move = 2, i2.end = 1, i2.cancel = 3, i2), a2[5] = (d2 = {}, d2.start = 2, d2.move = 2, d2.end = 1, d2.cancel = 3, d2), a2) }[Number(t)][e2];
  return l2 !== void 0 && l2[n2] || 0;
}
function X(t) {
  [1, 3, 2].includes(t.state) && (t.state = 0);
}
function Y(t) {
  return [4, 5].includes(t);
}
function k(t) {
  if (t.disabled)
    return t.state = 0, true;
}
function q(e2, n2) {
  return __assign(__assign(__assign({}, e2), n2), { state: 0, disabled: false });
}
function E() {
  var e2, n2, r2, i2, c2 = 0;
  return function(s2) {
    if (e2 = n2, s2 !== void 0) {
      var p = function(t, e3) {
        var n3 = t.phase, r3 = t.points, i3 = t.changedPoints, c3 = t.nativeEvent, s3 = r3.length, p2 = a$1 === n3, v3 = l === n3 && s3 === 0 || d$2 === n3, f3 = Date.now(), l$1 = b(r3) || b(i3), h2 = l$1.x, d2 = l$1.y, g2 = c3.currentTarget;
        return Object.assign(t, { id: e3, x: h2, y: d2, timestamp: f3, isStart: p2, isEnd: v3, pointLength: s3, currentTarget: g2, getOffset: function(t2) {
          t2 === void 0 && (t2 = g2);
          var e4 = t2.getBoundingClientRect();
          return { x: h2 - Math.round(e4.left), y: d2 - Math.round(e4.top) };
        } });
      }(s2, c2 = Number.MAX_SAFE_INTEGER > c2 ? ++c2 : 1);
      n2 = p;
      var v2 = p.isStart, f2 = p.pointLength;
      return v2 && (r2 = p, e2 = void 0, i2 = 1 < f2 ? p : void 0), __assign(__assign({}, p), { prevInput: e2, startMultiInput: i2, startInput: r2 });
    }
  };
}
function b(t) {
  var e2 = t.length;
  if (0 < e2) {
    if (e2 === 1) {
      var n2 = t[0], r2 = n2.clientX, i2 = n2.clientY;
      return { x: Math.round(r2), y: Math.round(i2) };
    }
    var o2 = t.reduce(function(t2, e3) {
      return t2.x += e3[c], t2.y += e3[u], t2;
    }, { x: 0, y: 0 });
    return { x: Math.round(o2.x / e2), y: Math.round(o2.y / e2) };
  }
}
function L(t, n2, r2, i2) {
  r2.target, r2.currentTarget, r2.type;
  var o2, a2 = __rest(r2, ["target", "currentTarget", "type"]);
  return document.createEvent ? (o2 = document.createEvent("HTMLEvents")).initEvent(t, i2 == null ? void 0 : i2.bubbles, i2 == null ? void 0 : i2.cancelable) : o2 = new Event(t, i2), Object.assign(o2, a2, { match: function() {
    return r2.targets && 0 < r2.targets.length && r2.targets.every(function(t2) {
      return o2.currentTarget.contains(t2);
    });
  } }), n2.dispatchEvent(o2);
}
var T = [g, j, w$1, x$1];
var w = { domEvents: { bubbles: true, cancelable: true }, preventDefault: function(t) {
  if (t.target && "tagName" in t.target) {
    var e2 = t.target.tagName;
    return !/^(?:INPUT|TEXTAREA|BUTTON|SELECT)$/.test(e2);
  }
  return false;
} }, x = function(e2) {
  function i2(n2, r2) {
    var i3, u2 = e2.call(this) || this;
    u2.__computeFunctionList = [], u2.__computeFunctionCreatorList = [], u2.__pluginContexts = [], u2.el = n2, u2.c = {}, u2.__options = __assign(__assign({}, w), r2);
    var c2, s2, h2, y$1, b2 = function(t) {
      var e3 = E();
      return function(n3) {
        var r3 = [], i4 = [];
        Array.from(n3.touches).forEach(function(e4) {
          var n4 = e4.clientX, o3 = e4.clientY, a2 = e4.target;
          (t == null ? void 0 : t.contains(a2)) && (r3.push(a2), i4.push({ clientX: n4, clientY: o3, target: a2 }));
        });
        var o2 = Array.from(n3.changedTouches).map(function(t2) {
          return { clientX: t2.clientX, clientY: t2.clientY, target: t2.target };
        });
        return e3({ phase: n3.type.replace("touch", ""), changedPoints: o2, points: i4, nativeEvent: n3, target: n3.target, targets: r3 });
      };
    }(u2.el), x2 = (s2 = false, h2 = null, y$1 = E(), function(t) {
      var e3, n3 = t.clientX, r3 = t.clientY, i4 = t.type, u3 = t.button, d2 = t.target, g2 = [{ clientX: n3, clientY: r3, target: d2 }];
      if (M === i4 && u3 === 0)
        h2 = d2, s2 = true, e3 = a$1;
      else {
        if (!s2)
          return;
        F === i4 ? e3 = i$1 : y === i4 && (g2 = [], e3 = l, s2 = false);
      }
      var m2 = c2 || [{ clientX: n3, clientY: r3, target: d2 }];
      if (c2 = [{ clientX: n3, clientY: r3, target: d2 }], e3 !== void 0)
        return y$1({ phase: e3, changedPoints: m2, points: g2, target: h2, targets: [h2], nativeEvent: t });
    });
    if (u2.__inputCreatorMap = ((i3 = {})[g] = b2, i3[j] = b2, i3[w$1] = b2, i3[x$1] = b2, i3[M] = x2, i3[F] = x2, i3[y] = x2, i3), u2.on("at:after", function(t) {
      var e3 = t.target, n3 = t.__type, r3 = u2.__options.domEvents;
      r3 && u2.el !== void 0 && e3 && (L(n3, e3, t, r3), L("at:after", e3, t, r3));
    }), n2 !== void 0) {
      n2.style.webkitTapHighlightColor = "rgba(0,0,0,0)";
      var C = false;
      try {
        var M$1 = {};
        Object.defineProperty(M$1, "passive", { get: function() {
          C = true;
        } }), window.addEventListener("_", function() {
        }, M$1);
      } catch (t) {
      }
      u2.on("u", function(t, e3, n3) {
        return T.forEach(function(r3) {
          t.addEventListener(r3, e3, n3);
        }), t.addEventListener(M, e3, n3), window.addEventListener(F, e3, n3), window.addEventListener(y, e3, n3), function() {
          T.forEach(function(n4) {
            t.removeEventListener(n4, e3);
          }), t.removeEventListener(M, e3, n3), window.removeEventListener(F, e3, n3), window.removeEventListener(y, e3, n3);
        };
      }(n2, u2.catchEvent.bind(u2), !(u2.__options.preventDefault !== false || !C) && { passive: true }));
    }
    return u2;
  }
  return __extends(i2, e2), i2.prototype.use = function(t, e3) {
    this.__pluginContexts.push(t(this, e3));
  }, i2.prototype.catchEvent = function(e3) {
    var n$1 = this.__inputCreatorMap[e3.type](e3);
    if (n$1 !== void 0) {
      var r2 = function() {
        return e3.preventDefault();
      };
      (function(t, e4) {
        var n$12 = e4.preventDefault;
        return n(n$12) ? n$12(t) : !!n$12;
      })(e3, this.__options) && r2(), this.emit("input", n$1), this.emit2("at:" + n$1.phase, n$1, {});
      var i3 = {};
      this.__computeFunctionList.forEach(function(t) {
        var e4 = t(n$1, i3);
        if (e4 !== void 0)
          for (var r3 in e4)
            i3[r3] = e4[r3];
      }), this.emit(m$1, __assign(__assign(__assign({}, n$1), i3), { stopPropagation: function() {
        return e3.stopPropagation();
      }, stopImmediatePropagation: function() {
        return e3.stopImmediatePropagation();
      }, preventDefault: r2 }));
    }
  }, i2.prototype.compute = function(t, e3) {
    var n2, i3;
    try {
      for (var o2 = __values(t), a2 = o2.next(); !a2.done; a2 = o2.next()) {
        var u2 = a2.value;
        this.__computeFunctionCreatorList.includes(u2) || (this.__computeFunctionCreatorList.push(u2), this.__computeFunctionList.push(u2()));
      }
    } catch (t2) {
      n2 = { error: t2 };
    } finally {
      try {
        a2 && !a2.done && (i3 = o2.return) && i3.call(o2);
      } finally {
        if (n2)
          throw n2.error;
      }
    }
    this.on(m$1, e3);
  }, i2.prototype.beforeEach = function(t) {
    var n2 = this;
    e2.prototype.beforeEach.call(this, function(e3, r2) {
      var i3;
      ((i3 = n2.c) === null || i3 === void 0 ? void 0 : i3.name) ? t(e3, r2) : r2();
    });
  }, i2.prototype.get = function(t) {
    return this.__pluginContexts.find(function(e3) {
      return t === e3.name;
    });
  }, i2.prototype.set = function(e3) {
    this.__options = __assign(__assign({}, this.__options), e3);
  }, i2.prototype.emit2 = function(e3, n2, r2) {
    var i3 = this;
    this.c = r2, this.emit(e3, __assign(__assign({}, n2), { type: e3 }), function() {
      i3.emit("at:after", __assign(__assign({}, n2), { name: e3, __type: e3 }));
    });
  }, i2.prototype.destroy = function() {
    this.emit("u"), e2.prototype.destroy.call(this);
  }, i2;
}(e$1);
var a = function(r2) {
  return Math.sqrt(r2.x * r2.x + r2.y * r2.y);
}, i = function(r2) {
  return r2 / Math.PI * 180;
}, h = function(a2, o2) {
  if (a2 !== 0 || o2 !== 0)
    return Math.abs(a2) >= Math.abs(o2) ? 0 < a2 ? v$1 : s : 0 < o2 ? h$1 : f$1;
};
function v() {
  return function(t) {
    var r2 = t.prevInput, a2 = 0, e2 = 0, o2 = 0;
    if (r2 !== void 0 && (a2 = t.x - r2.x, e2 = t.y - r2.y, a2 !== 0 || e2 !== 0)) {
      var i$12 = Math.sqrt(Math.pow(a2, 2) + Math.pow(e2, 2));
      o2 = Math.round(i(Math.acos(Math.abs(a2) / i$12)));
    }
    return { deltaX: a2, deltaY: e2, deltaXYAngle: o2 };
  };
}
function d$1() {
  var t, n2 = 0, c$1 = 0, p = 0, s2 = 0, v2 = 0;
  return function(d2) {
    var h$12 = d2.phase, f2 = d2.startInput;
    return a$1 === h$12 ? (n2 = 0, c$1 = 0, p = 0, s2 = 0, v2 = 0) : i$1 === h$12 && (n2 = Math.round(d2.points[0][c] - f2.points[0][c]), c$1 = Math.round(d2.points[0][u] - f2.points[0][u]), p = Math.abs(n2), s2 = Math.abs(c$1), v2 = Math.round(a({ x: p, y: s2 })), t = h(n2, c$1)), { displacementX: n2, displacementY: c$1, distanceX: p, distanceY: s2, distance: v2, overallDirection: t };
  };
}
function f() {
  var t, n2, r$1 = 0, e2 = 0, o2 = 0, i2 = 0;
  return function(u2) {
    if (u2 !== void 0) {
      n2 = n2 || u2.startInput;
      var c2 = u2.timestamp - n2.timestamp;
      if (r < c2) {
        var s2 = u2.x - n2.x, v2 = u2.y - n2.y;
        o2 = Math.round(s2 / c2 * 100) / 100, i2 = Math.round(v2 / c2 * 100) / 100, r$1 = Math.abs(o2), e2 = Math.abs(i2), t = h(s2, v2), n2 = u2;
      }
    }
    return { velocityX: r$1, velocityY: e2, speedX: o2, speedY: i2, direction: t };
  };
}
var d = { name: "pan", threshold: 10, pointLength: 1 };
function m(m2, u2) {
  var f$12 = q(d, u2);
  return m2.compute([f, d$1, v], function(t) {
    if (X(f$12), !k(f$12)) {
      var h2, c2, p, d2, u3, l$1 = (h2 = t.pointLength, c2 = t.distance, p = t.direction, d2 = t.phase, u3 = f$12.state, (Y(u3) || f$12.threshold <= c2) && f$12.pointLength === h2 && p !== void 0 || Y(u3) && [d$2, l].includes(d2));
      if (f$12.state = S(l$1, f$12.state, t.phase), l$1) {
        var v2 = f$12.name;
        m2.emit2(v2, t, f$12), m2.emit2(v2 + O(f$12.state), t, f$12), ![l, d$2].includes(t.phase) && t.direction && m2.emit2(v2 + t.direction, t, f$12);
      }
    }
  }), f$12;
}
var render = function() {
  var _vm = this;
  var _h = _vm.$createElement;
  var _c = _vm._self._c || _h;
  return _c("div", { staticClass: "audio-player" }, [_c("div", { staticClass: "audio__btn-wrap" }, [_vm.showPlaybackRate ? _c("div", { staticClass: "audio__play-rate", style: {
    color: _vm.themeColor
  } }, [_c("span", { on: { "click": function($event) {
    $event.stopPropagation();
    _vm.isShowRates = !_vm.isShowRates;
  } } }, [_vm._v(_vm._s(_vm.playbackRate.toFixed(1) + "x"))]), _c("transition", { attrs: { "name": "fade-rate" } }, [_c("div", { directives: [{ name: "show", rawName: "v-show", value: _vm.isShowRates, expression: "isShowRates" }], staticClass: "audio__play-rate__dropdown", style: {
    backgroundColor: _vm.themeColor
  } }, _vm._l(_vm.playbackRates, function(rate) {
    return _c("div", { key: "pr_" + rate, on: { "click": function($event) {
      $event.stopPropagation();
      return _vm.handleSetPlaybackRate(rate);
    } } }, [_vm._v(" " + _vm._s(rate.toFixed(1) + "x") + " ")]);
  }), 0)])], 1) : _vm._e(), _vm.showPrevButton ? _c("div", { staticClass: "audio__play-prev", class: { disable: !_vm.isLoop && _vm.currentPlayIndex === 0 }, style: {
    color: _vm.themeColor
  }, on: { "click": function($event) {
    $event.stopPropagation();
    return _vm.playPrev.apply(null, arguments);
  } } }, [_vm._t("play-prev", function() {
    return [_c("svg", { staticClass: "audio__play-icon", attrs: { "aria-hidden": "true" } }, [_c("use", { attrs: { "xlink:href": "#icon-play-prev" } })])];
  })], 2) : _vm._e(), _vm.isLoading ? _c("div", { staticClass: "audio__play-loading" }, _vm._l(8, function(item) {
    return _c("span", { key: item, style: {
      backgroundColor: _vm.themeColor
    } });
  }), 0) : [!_vm.isPlaying && _vm.showPlayButton ? _c("div", { staticClass: "audio__play-start", style: {
    color: _vm.themeColor
  }, on: { "click": function($event) {
    $event.stopPropagation();
    return _vm.play.apply(null, arguments);
  } } }, [_vm._t("play-start", function() {
    return [_c("svg", { staticClass: "audio__play-icon", attrs: { "aria-hidden": "true" } }, [_c("use", { attrs: { "xlink:href": "#icon-play" } })])];
  })], 2) : _vm.showPlayButton ? _c("div", { staticClass: "audio__play-pause", style: {
    color: _vm.themeColor
  }, on: { "click": function($event) {
    $event.stopPropagation();
    return _vm.pause.apply(null, arguments);
  } } }, [_vm._t("play-pause", function() {
    return [_c("svg", { staticClass: "audio__play-icon", attrs: { "aria-hidden": "true" } }, [_c("use", { attrs: { "xlink:href": "#icon-play-pause" } })])];
  })], 2) : _vm._e()], _vm.showNextButton ? _c("div", { staticClass: "audio__play-next", class: {
    disable: !_vm.isLoop && _vm.currentPlayIndex === _vm.audioList.length - 1
  }, style: {
    color: _vm.themeColor
  }, on: { "click": function($event) {
    $event.stopPropagation();
    return _vm.playNext.apply(null, arguments);
  } } }, [_vm._t("play-next", function() {
    return [_c("svg", { staticClass: "audio__play-icon", attrs: { "aria-hidden": "true" } }, [_c("use", { attrs: { "xlink:href": "#icon-play-next" } })])];
  })], 2) : _vm._e(), _vm.showVolumeButton ? _c("div", { staticClass: "audio__play-volume-icon-wrap" }, [_c("svg", { staticClass: "audio__play-icon", style: {
    color: _vm.themeColor
  }, attrs: { "aria-hidden": "true" }, on: { "click": function($event) {
    $event.stopPropagation();
    return _vm.handleVolumeIconTouchstart.apply(null, arguments);
  } } }, [_c("use", { attrs: { "xlink:href": _vm.currentVolume ? "#icon-play-volume" : "#icon-play-volume-no" } })]), _c("transition", { attrs: { "name": "fade-volume" } }, [_c("div", { directives: [{ name: "show", rawName: "v-show", value: _vm.isShowVolume, expression: "isShowVolume" }], ref: "playVolumeWrap", staticClass: "audio__play-volume-wrap", on: { "click": function($event) {
    $event.stopPropagation();
    return _vm.handleVolumePanmove.apply(null, arguments);
  }, "panmove": _vm.handleVolumePanmove, "panend": _vm.handleVolumePanend } }, [_c("div", { ref: "playVolume", staticClass: "audio__play-volume", style: {
    height: _vm.currentVolume * 100 + "%",
    backgroundColor: _vm.themeColor
  } })])])], 1) : _vm._e(), _c("div", { directives: [{ name: "show", rawName: "v-show", value: _vm.isShowErrorMessage, expression: "isShowErrorMessage" }], staticClass: "audio__notice" }, [_vm._v(" " + _vm._s(_vm.noticeMessage) + " ")])], 2), _c("div", { directives: [{ name: "show", rawName: "v-show", value: _vm.showProgressBar, expression: "showProgressBar" }], ref: "audioProgressWrap", staticClass: "audio__progress-wrap", on: { "click": function($event) {
    $event.stopPropagation();
    return _vm.handleClickProgressWrap.apply(null, arguments);
  } } }, [_c("div", { ref: "audioProgress", staticClass: "audio__progress", style: {
    backgroundColor: _vm.themeColor
  } }), _c("div", { ref: "audioProgressPoint", staticClass: "audio__progress-point", style: {
    backgroundColor: _vm.themeColor,
    boxShadow: "0 0 10px 0 " + _vm.themeColor
  }, on: { "panstart": _vm.handleProgressPanstart, "panend": _vm.handleProgressPanend, "panmove": _vm.handleProgressPanmove } })]), _c("div", { directives: [{ name: "show", rawName: "v-show", value: _vm.showProgressBar, expression: "showProgressBar" }], staticClass: "audio__time-wrap" }, [_c("div", { staticClass: "audio__current-time" }, [_vm._v(" " + _vm._s(_vm.currentTimeFormatted) + " ")]), _c("div", { staticClass: "audio__duration" }, [_vm._v(" " + _vm._s(_vm.durationFormatted) + " ")])]), _c("audio", _vm._b({ ref: "audio", staticClass: "audio-player__audio", attrs: { "src": _vm.audioList[_vm.currentPlayIndex] }, on: { "ended": _vm.onEnded, "timeupdate": _vm.onTimeUpdate, "loadedmetadata": _vm.onLoadedmetadata } }, "audio", _vm.$attrs, false), [_vm._v(" \u6D4F\u89C8\u5668\u592A\u8001\u54AF\uFF0C\u8BF7\u5347\u7EA7\u6D4F\u89C8\u5668\u5427~ ")])]);
};
var staticRenderFns = [];
var index_vue_vue_type_style_index_0_lang = "";
function normalizeComponent(scriptExports, render2, staticRenderFns2, functionalTemplate, injectStyles, scopeId, moduleIdentifier, shadowMode) {
  var options = typeof scriptExports === "function" ? scriptExports.options : scriptExports;
  if (render2) {
    options.render = render2;
    options.staticRenderFns = staticRenderFns2;
    options._compiled = true;
  }
  if (functionalTemplate) {
    options.functional = true;
  }
  if (scopeId) {
    options._scopeId = "data-v-" + scopeId;
  }
  var hook;
  if (moduleIdentifier) {
    hook = function(context) {
      context = context || this.$vnode && this.$vnode.ssrContext || this.parent && this.parent.$vnode && this.parent.$vnode.ssrContext;
      if (!context && typeof __VUE_SSR_CONTEXT__ !== "undefined") {
        context = __VUE_SSR_CONTEXT__;
      }
      if (injectStyles) {
        injectStyles.call(this, context);
      }
      if (context && context._registeredComponents) {
        context._registeredComponents.add(moduleIdentifier);
      }
    };
    options._ssrRegister = hook;
  } else if (injectStyles) {
    hook = shadowMode ? function() {
      injectStyles.call(this, (options.functional ? this.parent : this).$root.$options.shadowRoot);
    } : injectStyles;
  }
  if (hook) {
    if (options.functional) {
      options._injectStyles = hook;
      var originalRender = options.render;
      options.render = function renderWithStyleInjection(h2, context) {
        hook.call(context);
        return originalRender(h2, context);
      };
    } else {
      var existing = options.beforeCreate;
      options.beforeCreate = existing ? [].concat(existing, hook) : [hook];
    }
  }
  return {
    exports: scriptExports,
    options
  };
}
const __vue2_script = {
  name: "AudioPlayer",
  inheritAttrs: false,
  props: {
    audioList: {
      default: null,
      type: Array
    },
    showPlayButton: {
      default: true,
      type: Boolean
    },
    showPrevButton: {
      default: true,
      type: Boolean
    },
    showNextButton: {
      default: true,
      type: Boolean
    },
    showVolumeButton: {
      default: true,
      type: Boolean
    },
    showProgressBar: {
      default: true,
      type: Boolean
    },
    beforePlay: {
      default: null,
      type: Function
    },
    beforePrev: {
      default: null,
      type: Function
    },
    beforeNext: {
      default: null,
      type: Function
    },
    isLoop: {
      type: Boolean,
      default: true
    },
    isAutoPlayNext: {
      type: Boolean,
      default: true
    },
    progressInterval: {
      default: 1e3,
      type: Number
    },
    showPlaybackRate: {
      type: Boolean,
      default: true
    },
    playbackRates: {
      type: Array,
      default: () => [0.5, 1, 1.5, 2]
    },
    themeColor: {
      type: String,
      default: "#ff2929"
    },
    disabledProgressDrag: {
      type: Boolean,
      default: false
    },
    disabledProgressClick: {
      type: Boolean,
      default: false
    }
  },
  emits: [
    "pause",
    "play-prev",
    "play-next",
    "timeupdate",
    "loadedmetadata",
    "ended",
    "progress-start",
    "progress-end",
    "progress-move",
    "progress-click",
    "playing",
    "play",
    "play-error"
  ],
  data() {
    return {
      isIOS: /iPhone|iPad|iPod/i.test(window.navigator.userAgent),
      isPlaying: false,
      isDragging: false,
      isDraggingVolume: false,
      isShowErrorMessage: false,
      isLoading: false,
      isShowVolume: false,
      isShowRates: false,
      timer: null,
      noticeMessage: "",
      duration: "",
      currentPlayIndex: 0,
      currentTime: "",
      currentVolume: 1,
      playbackRate: 1,
      at: null
    };
  },
  computed: {
    currentTimeFormatted() {
      return this.currentTime ? this.formatTime(this.currentTime) : "00:00";
    },
    durationFormatted() {
      return this.duration ? this.formatTime(this.duration) : "00:00";
    }
  },
  mounted() {
    this.at = new x(this.$el, { preventDefault: false });
    this.at.use(m);
  },
  beforeUnmount() {
    this.at.destroy();
    this.pause();
  },
  methods: {
    handleVolumeIconTouchstart() {
      this.isShowVolume = !this.isShowVolume;
    },
    handleVolumePanmove(event) {
      let playVolumeWrapRect = this.$refs.playVolumeWrap.getBoundingClientRect();
      let pageY = event.y;
      let offsetTop;
      let volume;
      offsetTop = Math.round(playVolumeWrapRect.bottom - pageY);
      volume = offsetTop / this.$refs.playVolumeWrap.offsetHeight;
      volume = Math.min(volume, 1);
      volume = Math.max(volume, 0);
      this.$refs.audio.volume = volume;
      this.currentVolume = volume;
    },
    handleVolumePanend() {
      this.isShowVolume = false;
    },
    handleSetPlaybackRate(rate) {
      this.playbackRate = +rate;
      this.$refs.audio.playbackRate = +rate;
      this.isShowRates = false;
    },
    handleShowErrorMessage(opts = {}) {
      this.noticeMessage = opts.message;
      this.isShowErrorMessage = true;
      window.setTimeout(() => {
        this.isShowErrorMessage = false;
      }, opts.duration || 3e3);
    },
    onLoadedmetadata(event) {
      this.duration = this.$refs.audio.duration;
      this.$emit("loadedmetadata", event);
    },
    onTimeUpdate(event) {
      this.$emit("timeupdate", event);
    },
    formatTime(second) {
      let hour;
      hour = Math.floor(second / 60);
      second = Math.ceil(second % 60);
      hour += "";
      second += "";
      hour = hour.length === 1 ? "0" + hour : hour;
      second = second.length === 1 ? "0" + second : second;
      return hour + ":" + second;
    },
    onEnded(event) {
      window.setTimeout(() => {
        this.pause();
        this.$emit("ended", event);
        if (this.isLoop && this.isAutoPlayNext) {
          this.playNext();
        }
      }, 1e3);
    },
    handleProgressPanstart(event) {
      if (this.disabledProgressDrag)
        return;
      this.isDragging = true;
      this.$emit("progress-start", event);
    },
    handleProgressPanend(event) {
      if (this.disabledProgressDrag)
        return;
      this.$refs.audio.currentTime = this.currentTime;
      this.play();
      this.isDragging = false;
      this.$emit("progress-end", event);
    },
    handleProgressPanmove(event) {
      if (this.disabledProgressDrag)
        return;
      let pageX = event.x;
      let bcr = event.target.getBoundingClientRect();
      let targetLeft = parseInt(getComputedStyle(event.target).left);
      let offsetLeft = targetLeft + (pageX - bcr.left);
      offsetLeft = Math.min(offsetLeft, this.$refs.audioProgressWrap.offsetWidth);
      offsetLeft = Math.max(offsetLeft, 0);
      this.setPointPosition(offsetLeft);
      this.$refs.audioProgress.style.width = offsetLeft + "px";
      this.currentTime = offsetLeft / this.$refs.audioProgressWrap.offsetWidth * this.duration;
      this.$emit("progress-move", event);
    },
    handleClickProgressWrap(event) {
      if (this.disabledProgressClick)
        return;
      let target = event.target;
      let offsetX = event.offsetX;
      if (target === this.$refs.audioProgressPoint) {
        return;
      }
      this.currentTime = offsetX / this.$refs.audioProgressWrap.offsetWidth * this.duration;
      this.$refs.audio.currentTime = this.currentTime;
      this.setPointPosition(offsetX);
      this.$refs.audioProgress.style.width = offsetX + "px";
      this.play();
      this.$emit("progress-click", event);
    },
    setPointPosition(offsetLeft) {
      this.$refs.audioProgressPoint.style.left = offsetLeft - this.$refs.audioProgressPoint.offsetWidth / 2 + "px";
    },
    playing() {
      if (this.isDragging) {
        return;
      }
      let offsetLeft = this.$refs.audio.currentTime / this.$refs.audio.duration * this.$refs.audioProgressWrap.offsetWidth;
      this.currentTime = this.$refs.audio.currentTime;
      this.$refs.audioProgress.style.width = offsetLeft + "px";
      this.setPointPosition(offsetLeft);
      this.$emit("playing");
    },
    play() {
      this.isLoading = true;
      let handlePlay = () => {
        this.$refs.audio.play().then(() => {
          this.$nextTick(() => {
            this.clearTimer();
            this.timer = window.setInterval(this.playing, this.progressInterval);
            this.isPlaying = true;
            this.isLoading = false;
          });
          this.$emit("play");
        }).catch((data) => {
          this.handleShowErrorMessage({
            message: data.message
          });
          if (data.code === 9) {
            if (this.isAutoPlayNext) {
              window.setTimeout(() => {
                this.playNext();
              }, 1e3);
            }
          }
          this.isLoading = false;
          this.$emit("play-error", data);
        });
      };
      if (this.isIOS) {
        console.log("\u4E3A\u4E86\u89E3\u51B3 iOS \u8BBE\u5907\u63A5\u53E3\u5F02\u6B65\u8BF7\u6C42\u540E\u51FA\u73B0\u65E0\u6CD5\u64AD\u653E\u95EE\u9898\uFF0C\u8BF7\u65E0\u89C6 The play() request was interrupted by a call to pause() \u9519\u8BEF");
        this.$refs.audio.play();
        this.$refs.audio.pause();
      }
      if (this.beforePlay) {
        this.beforePlay((state) => {
          if (state !== false) {
            handlePlay();
          }
        });
        return;
      }
      handlePlay();
    },
    pause() {
      this.$refs.audio.pause();
      this.$nextTick(() => {
        this.clearTimer();
        this.isPlaying = false;
        this.$emit("pause");
      });
    },
    playPrev() {
      if (this.currentPlayIndex <= 0 && !this.isLoop) {
        return;
      }
      this.clearTimer();
      let handlePrev = () => {
        if (this.currentPlayIndex <= 0 && this.isLoop) {
          this.currentPlayIndex = this.audioList.length - 1;
        } else {
          this.currentPlayIndex--;
        }
        this.$nextTick(() => {
          this.play();
          this.$emit("play-prev");
        });
      };
      if (this.beforePrev) {
        this.beforePrev((state) => {
          if (state !== false) {
            handlePrev();
          }
        });
        return;
      }
      handlePrev();
    },
    clearTimer() {
      window.clearInterval(this.timer);
      this.timer = null;
    },
    playNext() {
      if (this.currentPlayIndex + 1 >= this.audioList.length && !this.isLoop) {
        return;
      }
      this.clearTimer();
      let handleNext = () => {
        if (this.currentPlayIndex + 1 >= this.audioList.length && this.isLoop) {
          this.currentPlayIndex = 0;
        } else {
          this.currentPlayIndex++;
        }
        this.$nextTick(() => {
          this.play();
          this.$emit("play-next");
        });
      };
      if (this.beforeNext) {
        this.beforeNext((state) => {
          if (state !== false) {
            handleNext();
          }
        });
        return;
      }
      handleNext();
    }
  }
};
const __cssModules = {};
var __component__ = /* @__PURE__ */ normalizeComponent(__vue2_script, render, staticRenderFns, false, __vue2_injectStyles, null, null, null);
function __vue2_injectStyles(context) {
  for (let o2 in __cssModules) {
    this[o2] = __cssModules[o2];
  }
}
var AudioPlayer = /* @__PURE__ */ function() {
  return __component__.exports;
}();
AudioPlayer.install = (Vue) => {
  Vue.component(AudioPlayer.name, AudioPlayer);
};
export { AudioPlayer as default };
